packages/devtools_app/lib/src/screens/memory/shared/primitives/instance_set_view.dart

```diff
@@ -0,0 +1,95 @@
+// Copyright 2023 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:intl/intl.dart';
+import 'package:vm_service/vm_service.dart';
+
+import '../../../../shared/analytics/constants.dart';
+import '../../../../shared/common_widgets.dart';
+
+typedef SampleObtainer = InstanceRef Function();
+
+class InstanceSetView extends StatelessWidget {
+  const InstanceSetView({
+    super.key,
+    required this.count,
+    required this.sampleObtainer,
+    required this.showMenu,
+    this.textStyle,
+    required this.gaContext,
+  }) : assert(showMenu == (sampleObtainer != null));
+
+  final int count;
+  final SampleObtainer? sampleObtainer;
+  final bool showMenu;
+  final TextStyle? textStyle;
+  final MemoryAreas gaContext;
+
+  @override
+  Widget build(BuildContext context) {
+    final format = NumberFormat.decimalPattern();
+
+    return Row(
+      children: [
+        Text(
+          format.format(count),
+          style: textStyle,
+        ),
+        if (showMenu)
+          ContextMenuButton(
+            style: textStyle,
+            menu: _menu(),
+          ),
+        if (!showMenu) const SizedBox(width: ContextMenuButton.width),
+      ],
+    );
+  }
+}
+
+List<Widget> _menu() => [
+      SubmenuButton(
+        menuChildren: <Widget>[
+          MenuItemButton(
+            onPressed: () => print('get fields'),
```

Comment:
Is there any way we can make this constant? I assume this closure is what's preventing this entire `List` from being constant.

Reply:
the menu items are going to have real functional handlers, so no chances it will stay constant